server/base/src/main/java/org/apache/accumulo/server/conf/util/ZooPropSetTool.java

```diff
@@ -0,0 +1,391 @@
+    if (!opts.tableOpt.isEmpty() || !opts.tableIdOpt.isEmpty()) {
+      TableId tid = getTableId(iid, opts, zooReader);
+      return TablePropKey.of(iid, tid);
+    }
+
+    // either nid of namespace name provided, get the namespace id.
+    if (!opts.namespaceOpt.isEmpty() || !opts.namespaceIdOpt.isEmpty()) {
+      NamespaceId nid = getNamespaceId(iid, opts, zooReader);
+      return NamespacePropKey.of(iid, nid);
+    }
+
+    // no table or namespace, assume system.
+    return SystemPropKey.of(iid);
+  }
+
+  private TableId getTableId(final InstanceId iid, final ZooPropSetTool.Opts opts,
+      final ZooReader zooReader) {
+    if (!opts.tableIdOpt.isEmpty()) {
+      return TableId.of(opts.tableIdOpt);
+    }
+    Map<NamespaceId,String> nids = getNamespaceIdToNameMap(iid, zooReader);
+
+    Map<TableId,String> tids = getTableIdToName(iid, nids, zooReader);
+    return tids.entrySet().stream().filter(entry -> opts.tableOpt.equals(entry.getValue()))
+        .map(Map.Entry::getKey).findAny()
+        .orElseThrow(() -> new IllegalArgumentException("Could not find table " + opts.tableOpt));
+  }
+
+  private NamespaceId getNamespaceId(final InstanceId iid, final ZooPropSetTool.Opts opts,
+      final ZooReader zooReader) {
+    if (!opts.namespaceIdOpt.isEmpty()) {
+      return NamespaceId.of(opts.namespaceIdOpt);
+    }
+    Map<NamespaceId,String> nids = getNamespaceIdToNameMap(iid, zooReader);
+    return nids.entrySet().stream().filter(entry -> opts.namespaceOpt.equals(entry.getValue()))
+        .map(Map.Entry::getKey).findAny().orElseThrow(
+            () -> new IllegalArgumentException("Could not find namespace " + opts.namespaceOpt));
+  }
+
+  static class Opts extends ConfigOpts {
+
+    @Parameter(names = {"-d", "--delete"}, description = "delete a property")
+    public String deleteOpt = "";
+    @Parameter(names = {"-f", "--filter"},
+        description = "show only properties that contain this string in their name.")
+    public String filterOpt = "";
+    @Parameter(names = {"-fv", "--filter-with-values"},
+        description = "show only properties that contain this string in their name.")
+    public String filterWithValuesOpt = "";
+    @Parameter(names = {"--instanceName"},
+        description = "Specify the instance name to use. If instance name or id are not provided, determined from configuration (requires a running hdfs instance)")
+    public String instanceName = "";
+    @Parameter(names = {"--instanceId"},
+        description = "Specify the instance id to use. If instance name or id are not provided, determined from configuration (requires a running hdfs instance)")
+    public String instanceId = "";
+    @Parameter(names = {"-ns", "--namespace"},
+        description = "namespace to display/set/delete properties for")
+    public String namespaceOpt = "";
+    @Parameter(names = {"-nid", "--namespace-id"},
+        description = "namespace id to display/set/delete properties for")
```

Comment:
I would keep it simple and only have the namespace name. We can add an "are you sure" (and a `-f, --force` bypass option) prompt that displays the namespace ID that was resolved for the given name.

Same comment for table name and table ID. We only need the name, just like in the shell.

Any view options should display the name and id for these, though.

Reply:
Allowing and id or a name was to ease using information if it was coming from a log file.  Often, the logs use id, not name.  Making the user dig out the name, potentially without the shell being available seemed like an unnecessary step when the tool could use either.  Internally, the code uses id and if given a name it takes effort to lookup the id, so just accepting the id that is needed anyway didn't seem to add more complexity.

Reply:
I'm convinced. Having both is good. You're right that it doesn't really add complexity.